src/com/palmergames/bukkit/towny/db/TownyFlatFileSource.java

```diff
@@ -1521,6 +1505,9 @@ public boolean loadTownBlocks() {
 						try {
 							Town town = getTown(line.trim());
 							townBlock.setTown(town);
+							TownyWorld townyWorld = townBlock.getWorld();
+							if (townyWorld != null && !townyWorld.hasTown(town))
+								townyWorld.addTown(town);
```

Comment:
Taking into consideration runtime, you're going through every loaded town for `!townyWorld.hasTown(town)`, one recommendation to solve this problem would be using a `HashMap<String, Town>` in `TownyWorld` to store towns as the fetch runtime for this data structure is O(1) which is much faster than .contains()